pkg/system/phase4_configuring.go

```diff
@@ -1323,6 +1327,49 @@ func (r *Reconciler) ReconcileOBCStorageClass() error {
 	return nil
 }
 
+// ReconcileBucketLoggingPVC ensures the bucket logging PVC is properly configured for 'guaranteed' logging
+func (r *Reconciler) ReconcileBucketLoggingPVC() error {
+	// Return if logging is not enabled or if 'guaranteed' logging is not specified
+	if r.NooBaa.Spec.BucketLogging.LoggingType != nbv1.BucketLoggingTypeGuaranteed {
+		return nil
+	}
+
+	if err := r.prepareBucketLoggingPVC(); err != nil {
+		return err
+	}
+
+	r.Own(r.BucketLoggingPVC)
+	// TODO: return only for the specific error, like RWX mode not supported by provisioner
+	if err := r.Client.Create(r.Ctx, r.BucketLoggingPVC); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// prepareBucketLoggingPVC configures the bucket logging pvc
+func (r *Reconciler) prepareBucketLoggingPVC() error {
+	r.BucketLoggingPVC.Spec.AccessModes = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteMany}
+	r.BucketLoggingPVC.Spec.StorageClassName = r.NooBaa.Spec.BucketLogging.BucketLoggingStorageClass
+
+	if r.NooBaa.Spec.BucketLogging.BucketLoggingStorageClass == nil {
+		sc := &storagev1.StorageClass{
+			TypeMeta:   metav1.TypeMeta{Kind: "StorageClass"},
+			ObjectMeta: metav1.ObjectMeta{Name: "ocs-storagecluster-cephfs"},
+		}
+
+		// fallback to cephfs storageclass if running on odf and BucketLoggingStorageClass not provided
+		if util.KubeCheck(sc) {
+			r.BucketLoggingPVC.Spec.StorageClassName = &sc.Name
```

Comment:
Please add a warning here to the logs that we are using ceph-fs. Something like: "BucketLoggingStorageClass wasn't provided - defaulting to ceph-fs storage class"

Reply:
Sure, I will add. Thanks